### DISCUSSION TITLE
OIDC Authentication Configuration with Keycloak

### DISCUSSION THREAD

**Discussion by dayres** [ngoduykhanh/PowerDNS-Admin#1544] [Q&A]
I am configuring PDNS-Admin to authenticates against keycloak. Everything goes well until it redirects to /login to complete the process. I receive the following trace:

```
[2023-04-24 21:49:15 +0000] [7] [ERROR] Exception on /login [GET]
powerdns_admin | Traceback (most recent call last):
powerdns_admin |   File "/usr/lib/python3.10/site-packages/flask/app.py", line 2073, in wsgi_app
powerdns_admin |     response = self.full_dispatch_request()
powerdns_admin |   File "/usr/lib/python3.10/site-packages/flask/app.py", line 1519, in full_dispatch_request
powerdns_admin |     rv = self.handle_user_exception(e)
powerdns_admin |   File "/usr/lib/python3.10/site-packages/flask/app.py", line 1517, in full_dispatch_request
powerdns_admin |     rv = self.dispatch_request()
powerdns_admin |   File "/usr/lib/python3.10/site-packages/flask/app.py", line 1503, in dispatch_request
powerdns_admin |     return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
powerdns_admin |   File "/app/powerdnsadmin/routes/index.py", line 395, in login
powerdns_admin |     user_data = json.loads(oidc.get('userinfo').text)
powerdns_admin |   File "/usr/lib/python3.10/json/__init__.py", line 346, in loads
powerdns_admin |     return _default_decoder.decode(s)
powerdns_admin |   File "/usr/lib/python3.10/json/decoder.py", line 337, in decode
powerdns_admin |     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
powerdns_admin |   File "/usr/lib/python3.10/json/decoder.py", line 355, in raw_decode
powerdns_admin |     raise JSONDecodeError("Expecting value", s, err.value) from None
powerdns_admin | json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

Any insight or suggestions would be greatly appreciated. I'm unsure if this is a bug or a misconfiguration at this point.

**Comment by AzorianMatt** [comment]
It may be a little bit of both.

You might find yourself hard-pressed to make this work with Keycloak currently anyhow when considering this open known issue: https://github.com/PowerDNS-Admin/PowerDNS-Admin/issues/1222

**Reply by dayres** [reply]
Thanks for the quick response, I didn't see that in the issues for some reason. Luckily, I can use Azure for now as a workaround.

FYI, we are currently running Keycloak 21.0.1 and plan to upgrade to the latest 21.1.0 .

As our issue is during the login, do you think it is a configuration problem? I'd like to see if I could resolve it even if logout is not currently functioning.

**Reply by AzorianMatt** [reply]
@dayres You're welcome.

I do wonder if it's a configuration problem. I ponder this purely because of issue #1222 implying that it got far enough in to find an issue with the logout process. The PDA code for this certainly isn't the most robust, so either is quite plausible. It's my hope to get additional authentication testing environments deployed this year to aid in achieving feature stability for this PDA edition.
